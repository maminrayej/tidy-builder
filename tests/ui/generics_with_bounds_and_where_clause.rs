#[derive(tidy_builder::Builder)]
struct MyStruct<T: std::fmt::Debug>
where
    T: std::fmt::Display,
{
    req1: T,
    req2: T,
    opt1: Option<T>,
}

fn main() {
    let my_struct = MyStruct::builder()
        .opt1("opt1".to_string())
        .req2("req2".to_string())
        .build();

    assert_eq!(my_struct.req1, "req1".to_string());
    assert_eq!(my_struct.req2, "req2".to_string());
    assert_eq!(my_struct.opt1, Some("opt1".to_string()));
}
