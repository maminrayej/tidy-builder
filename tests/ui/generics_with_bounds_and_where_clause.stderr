error[E0277]: the trait bound `MyStructBuilder<false, true, String>: HasReq1` is not satisfied
  --> tests/ui/generics_with_bounds_and_where_clause.rs:15:10
   |
15 |         .build();
   |          ^^^^^ the trait `HasReq1` is not implemented for `MyStructBuilder<false, true, String>`
   |
   = help: the trait `HasReq1` is implemented for `MyStructBuilder<true, P1, T>`
note: required by a bound in `MyStructBuilder::<P0, P1, T>::build`
  --> tests/ui/generics_with_bounds_and_where_clause.rs:1:10
   |
1  | #[derive(tidy_builder::Builder)]
   |          ^^^^^^^^^^^^^^^^^^^^^ required by this bound in `MyStructBuilder::<P0, P1, T>::build`
   = note: this error originates in the derive macro `tidy_builder::Builder` (in Nightly builds, run with -Z macro-backtrace for more info)
